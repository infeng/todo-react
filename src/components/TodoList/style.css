ul, li {
  list-style: none;
  text-align: left;
}
#todo-list {
  margin-bottom: 20px;
}
#todo-list {
  border-top: 1px solid #e6e6e6;
  border-bottom: 1px solid #e6e6e6;
}
.no-todo-list {
  text-align: center;
  color: #d9d9d9;
}
