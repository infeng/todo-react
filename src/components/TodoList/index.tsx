import * as React from 'react';
import Todo from '../Todo';
import { TodoModel } from '../../reducers';

import './style.css';

export interface TodoModel {
  text: string;
  completed: boolean;
}

interface Props {
  onTodoClick: (index: number) => void;
  todos: TodoModel[];
}

export class TodoList extends React.Component<Props, any> {
  render() {
    let todosNode: any = this.props.todos.map((todo, index) => {
      return (
        <Todo
        key={index}
        onClick={() => this.props.onTodoClick(index)}
        text={todo.text}
        completed={todo.completed}
        isLast={index === (this.props.todos.length - 1) ? true : false}
        /> 
      );
    });
    if (this.props.todos.length === 0) {
      todosNode = <p className="todo no-todo-list">Nothing</p>;
    }
    return (
      <ul id="todo-list">
        {todosNode}
      </ul>
    );
  }
}
