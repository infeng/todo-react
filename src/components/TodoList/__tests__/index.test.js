import React from 'react';
import { render, shallow } from 'enzyme';
import TodoList from '..';

function createTodoList(props) {
  return (
    <TodoList {...props}/>
  );
}

describe('Footer', () => {
  it('render Footer correctly', () => {
    let todoList = createTodoList({
      todos: [
        {
          text: 'hello world',
          completed: false,
        },
        {
          text: '233',
          completed: false,
        }
      ],
    });

    let wrapper = render(todoList);
    expect(wrapper).toMatchSnapshot();
  });

  it('render correctly when no todos', () => {
    let todoList = createTodoList({
      todos: []
    });

    let wrapper = render(todoList);
    expect(wrapper).toMatchSnapshot();    
  });

  it('click todo', () => {
    let onTodoClick = jest.fn();

    let todoList = createTodoList({
      todos: [
        {
          text: 'hello world',
          completed: false,
        },
        {
          text: '233',
          completed: false,
        }
      ],
      onTodoClick,
    });

    let wrapper = shallow(todoList);
    wrapper.find('Todo').at(0).simulate('click');
    expect(onTodoClick).toBeCalledWith(0);
  });
});