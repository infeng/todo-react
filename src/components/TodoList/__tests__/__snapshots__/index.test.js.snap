// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render Footer correctly 1`] = `
<ul
  id="todo-list"
>
  <li
    class="todo notlast"
  >
    hello world
  </li>
  <li
    class="todo"
  >
    233
  </li>
</ul>
`;

exports[`Footer render correctly when no todos 1`] = `
<ul
  id="todo-list"
>
  <p
    class="todo no-todo-list"
  >
    Nothing
  </p>
</ul>
`;
